.gitignore

```diff
@@ -109,30 +109,9 @@ src/thrift/*
 !src/thrift/Makefile
 
 # Autogenerated Dockerfiles
-sonic-slave/Dockerfile
-sonic-slave-stretch/Dockerfile
-sonic-slave-jessie/Dockerfile
-dockers/docker-base/Dockerfile
-dockers/docker-base-stretch/Dockerfile
-dockers/docker-config-engine/Dockerfile
-dockers/docker-config-engine-stretch/Dockerfile
-dockers/docker-database/Dockerfile
-dockers/docker-dhcp-relay/Dockerfile
-dockers/docker-fpm-frr/Dockerfile
-dockers/docker-fpm-gobgp/Dockerfile
-dockers/docker-fpm-quagga/Dockerfile
-dockers/docker-lldp-sv2/Dockerfile
-dockers/docker-orchagent/Dockerfile
-dockers/docker-platform-monitor/Dockerfile
-dockers/docker-router-advertiser/Dockerfile
-dockers/docker-snmp-sv2/Dockerfile
-dockers/docker-teamd/Dockerfile
-dockers/docker-sonic-mgmt/Dockerfile
-dockers/docker-sonic-telemetry/Dockerfile
-platform/*/docker-saiserver-*/Dockerfile
-platform/*/docker-syncd-*/Dockerfile
-platform/*/docker-syncd-*-rpc/Dockerfile
-platform/vs/docker-sonic-vs/Dockerfile
+sonic-slave*/Dockerfile
+dockers/*/Dockerfile
+platform/*/docker-*/Dockerfile
```

Comment:
**/Dockerfiles?

Reply:
Not all of our Dockerfiles are generated from templates. We have some Dockerfiles checked in. I prefer to explicitly ignore generated ones, at least for now.

Reply:
@jleveque do you want to stick with explicitly listing each path or do you think it's OK to wild-card the directories where we know the Dockerfile is generated by the build?

Reply:
I think it's OK to wildcard specific directories, just not a global `**/Dockerfile`, unless we can ensure that all of the Dockerfiles in our repo can be and are templated (there aren't many left, actually).

On that note, I noticed you renamed the barefoot and cavium saiserver Dockerfiles as .j2 files. However, they contain no templating. I would prefer if they were also converted to templates in order to utilize the build system like similar templated Dockerfiles. However, I'm more concerned that these templates will not be used to generate files because they don't appear to have build targets associated with them. have you confirmed simply renaming these files doesn't break builds for those ASIC platforms?

Reply:
>However, I'm more concerned that these templates will not be used to generate files because they don't appear to have build targets associated with them.

Good catch! I think for now I'll just remove those renamings from this PR until we can verify how they're being used and/or convert them to templates. I've seen some other .j2s without templating which is why I updated them, but I'd rather be safe than sorry. :)